CMakeLists.txt

```diff
@@ -518,15 +518,6 @@ find_package(Boost 1.61 COMPONENTS ${BOOST_COMPONENTS} REQUIRED)
 include_directories(SYSTEM ${Boost_INCLUDE_DIRS})
 include_directories(SYSTEM ${PROJECT_BINARY_DIR}/include)
 
-if (WITH_SYSTEM_BOOST)
```

Comment:
Was this necessary for turning libcommon into a shared library, or is just a cleanup? If it's just to clean up, maybe we should separate it from this PR.